Fixes a circular dependency and manifest cleanup.

jtkech opened the pull request:
Fixes #1417 

- To fix the **circular deps** `ContentTypes` => `Contents` => `Liquid` => `Contents and ContentTypes` in **modules manifests**. Here, in the `Liquid` module, we make two separate features, one for `Liquid` which depends on nothing, and one for `LiquidPart` which depends on `Contents, ContentTypes, Liquid`.

- **Warning**: I updated our recipes. Maybe a breaking change for site which already use `Liquidpart`.

I also did some **manifest cleanups** because of the following.

- The `FeatureDescription` key doesn't exists so it is never shown up, so i removed the remaining ones. Already done in another PR but i will revert it and keep this change only here.

- When a module has a `Features` section, the properties defined at the module level, e.g a `Description` or a `Category`, are never used. So i removed them and e.g, when a 1st feature description was less descriptive than the unused module description, i used the module one.

Maybe todo

- There are **unnecessary deps**, e.g by using `OC.Contents, OC.ContentTypes, OC.Liquid` in some modules knowing that `OC.ContentTypes` depends on `OC.Contents` and so on.

- But maybe it is intentional to show up more dependencies for a given feature. This because it seems that only **direct dependencies** are shown, not the **full graph**.

So, let me know if it is worth to display, in the admin `Modules` page, all dependencies (not only direct ones) for each feature. Then, we could do more cleanups in modules manifest files.

sebastienros commented:
I don't think we should have feature for the LiquidPart. Just create a specific `Startup` that has a `[RequiredFeature]`. This way it will be registered only if the Contents feature is available, but it's not a required feature, or even a dependency.

If the migration needs it too, ensure that this attribute also handles migrations.

jtkech commented:
Yes, that's the right solution 👍 Why i didn't think about it? I will update it soon.

jtkech commented:
@sebastienros 

- I used `[RequireFeatures("OrchardCore.ContentTypes")]` for the custom `LiquidPart` startup.  Because `ContentTypes` depends on `Contents`, i didn't also add `Contents`.

- Then, because in this custom startup there is `services.AddScoped<IDataMigration, Migrations>()`. I think **we don't need to also decorate the migration class itself. Right**?

- Then, to break the circular dependencies, i removed the dependency of `Contents` on `Liquid`. This would mean that `Contents` could be enabled without `Liquid`. Seems ok because in `Contents` you also have a custom startup which requires `Liquid` to register a filter (**i added another one**).

- **Warning**: But not sure about all other modules which have a dependency on `Liquid`. Ok, this means that `Liquid` will be enabled if one is enabled. But this forces e.g the part drivers ordering just because they need `Liquid` services, not because they depend somewhere on the `LiquidPart`.

---

- I also fixed the `Modules` page which was only showing direct dependencies, **so now for each feature the full graph of its dependencies is displayed***.

- Then, in manifest files i removed many unnecessary deps (maybe not all), those which are not direct deps, but deducted from the graph. But, because of the fix they are still shown up in the `Modules` page.

- **Important**: I first remove all the `Contents` dependency when `ContentTypes` was also specified because it depends on `Contents`. When only `Contents` was specified, i replaced it with `ContentTypes` when the related module define a `ContentPart`, but not sure at 100% so please review.

-  **Important**: There was no dependencies for `OrchardCore.Title` so, because it defines a `ContentPart`, i added the `ContentTypes` dependency.

- **Infos**: Not important for me, but just saw that in the script the var `confirmDisableMessage` initialized with `Disabling this feature will also disable the following dependent features` is never used.

jtkech commented:
Not so easy to see which modules need a dependency on `ContentTypes`. First i looked up those which are defining `ContentTypes` services, but they are declared in `OC.ContentTypes.Abstractions`. So, maybe just when it doesn't make sense e.g to use a part without having the ability to define type part settings.

- So **i reverted back to the original dependencies** so that it is as before. But i still kept removed those that can be deducted from the full graph.

- Then i will re-think about the circular dependencies.

jtkech commented:
Humm, i think i got it.

- It seems that no module consume a service registered by `ContentTypes`, they just define services that are consumed by `ContentTypes` with interfaces defined in `.Abstractions`. So it seems that the dependency on `ContentTypes` is not required, but just used when it doesn't make sense to not have it.

- **So, for the `Liquid` part i think that's ok that it doesn't depend on `ContentTypes`**.

- `Body` defines a driver / handler which consumes services registered by `Contents`. But `Title` defines a driver / handler which doesn't consume a `Contents` service. So that's ok that `Title` depends on nothing. But e.g the `Liquid` part need a `Contents` service.

- Then there is the driver / handler ordering, but if `Title` doens't need to run after anyone, that's ok. And i think it is the same for the `LiquidPart`.

- **So, and because the `Liquid` part consume a `Contents` service, as you was saying, i think that's ok to just use a `RequireFeatures` on `Contents`**. I will update it.

- Then because we can re-introduce the `Contents` dependency on `Liquid` i will remove it in other modules which was depending both on `Contents` and `Liquid`.

sebastienros commented:
Just to be clear, a `[RequireFeatures]` usage doesn't imply that you need to add a feature dependency. Quite the contrary.

jtkech commented:
@sebastienros 

I think i'm ok, so as a summary

- Here we preserve original manifest dependencies, we just removed those which are deducted from the full graph. But now, in the `Modules` page, all dependencies are displayed, not only direct ones.

- To break the circular dependencies here we removed the `Liquid` dependencies on `ContentTypes and Contents`, and now we use a custom startup with `RequireFeatures("OC.ContentTypes")`. Here i think that i could have used `Contents`, but i didn't do it just because before it was depending on `ContentTypes`.

- Finally, i didn't decorate `Migrations` because the related service is registered in the custom startup.

jtkech commented:
@sebastienros just saw your comment

> Just to be clear, a [RequireFeatures] usage doesn't imply that you need to add a feature dependency. Quite the contrary.

Yes i know. I was just thinking at some point of the drivers / handlers ordering which only rely on real dependencies. So, no problem, the feature doesn't need to be enabled, but some services require this feature to be registered. But if you would want to run after this feature (when enabled) you can't force it through the `RequireFeatures`.

jtkech commented:
@sebastienros 

**But now all is ok** the `Liquid` part just "requires" `Contents`, and its driver / handler don't need to be run after any other, and this order doesn't depend on having or not a real dependency on `Contents`.

Sorry, i was not sure of all the details at the beginning. So, as a last summary.

- Fix the circular dependencies by removing the `Liquid` dependency on `ContentTypes` and `Contents`. Then add a custom startup with `[RequireFeatures("OrchardCore.Contents")]`.

- Remove unnecessary dependencies which are deducted from the full graph. I just added some `ContentTypes` for modules which provide services related to types settings.

- Now, in the `Modules` page, for each feature the full graph of its dependencies is displayed.

sebastienros commented on src/OrchardCore.Modules/OrchardCore.Features/Controllers/AdminController.cs:
```diff
@@ -62,16 +62,19 @@ public class AdminController : Controller
             {
-                var dependentFeatures = _extensionManager
-                    .GetDependentFeatures(moduleFeatureInfo.Id);
+                //var dependentFeatures = _extensionManager
```
remove comments
